src/Orleans.Core.Abstractions/Streams/Core/StreamSequenceToken.cs

```diff
@@ -10,6 +10,16 @@ namespace Orleans.Streams
     [Serializable]
     public abstract class StreamSequenceToken : IEquatable<StreamSequenceToken>, IComparable<StreamSequenceToken>
     {
+        /// <summary>
+        /// Number of event batches in stream prior to this event batch
+        /// </summary>
+        public long SequenceNumber { get; protected set; }
```

Comment:
what does this mean to other persistent stream providers?

Reply:
It means that ordering information needs be expressed in a long value.

This could be a problem if systems have more complex or convoluted ordering information.

It could also be an issue if provider implementations have persisted tokens that don't fit this.  They may need to perform some translation between the old token and a new compatible one.  This should not be an issue for any we maintain, but custom tokens may encounter some issues.